shotover/src/transforms/kafka/sink_cluster.rs

```diff
@@ -171,12 +181,56 @@ impl Transform for KafkaSinkCluster {
                     ..
                 })) => {
                     for broker in &mut metadata.brokers {
+                        let mut found = false;
+                        for node in &mut self.nodes {
+                            if broker.1.host == node.kafka_address.host
+                                && broker.1.port == node.kafka_address.port
+                            {
+                                found = true;
+                                node.broker_id = **broker.0;
+                            }
+                        }
+                        if !found {
+                            let host = broker.1.host.clone();
+                            let port = broker.1.port;
+                            self.nodes.push(KafkaNode {
+                                broker_id: **broker.0,
+                                address: format!("{host}:{port}").parse().map_err(|_| {
+                                    anyhow!(
+                                        "Failed to parse address from kafka message: {host}:{port}"
+                                    )
+                                })?,
+                                kafka_address: KafkaAddress { host, port },
+                                connection: None,
+                            });
+                        }
                         rewrite_address(
                             &self.shotover_nodes,
                             &mut broker.1.host,
                             &mut broker.1.port,
                         )
                     }
+
+                    for topic in &metadata.topics {
+                        self.topics.insert(
```

Comment:
This could _potentially_ grow to a significant size over the course of a long enough connection with sufficient topics / changes to topics? 

Is it worth clearing out old / stale entries based on the new metadata? Or is this a problem only in theory?

Reply:
the metadata request specifies a topic to retrieve metadata on, so we cant use a metadata response to determine which topics no longer exist.

I think the best approach would be to remove entries if a metadata response returns whatever error indicates that the topic does not exist. This might not be sufficient though.

We could implement `self.topics` with some kind of LRU eviction.
But that would mean we have to fetch metadata on demand if we evict something that the client expects to still be there.

Reply:
Ok thanks for the clarification I think just for the moment maybe evicting on the error would be sufficient - It will get cleared when a connection drops as well.

Reply:
huh, turns out its a bit more complicated than that.
if kafka is configured with `auto.create.topics.enable=true` (the default setting) then it will return error code 3 but also immediately create the topic.

We could determine if the server is configured with auto.create.topics.enable=false` and then in that case use error code 3 to clear `self.topics` values.
But considering it doesnt even help us in the default config it doesnt feel worth it.

I think for now lets rely on closing the connection to clear the cache, and then we can investigate alternatives if the memory growth turns out to be too large.